draft-ietf-dnsop-dns-catalog-zones.md

```diff
@@ -460,53 +458,40 @@ zones), but nameservers MUST NOT process such broken zones as catalog
-
-## Member zone name clash
-
-## Member zone removal
-
-When zones are deleted from a catalog zone, a primary MAY delete the member
-zone immediately.
-It is up to the secondary nameserver to handle this condition correctly.
+## Member zone removal {#zoneremoval}
 
-## Migrating zones between catalogs
+When a member zone is removed from a specific catalog zone, an authoritative server MUST NOT remove the zone and associated state data if the zone was not configured from that specific catalog zone.
+Only when the zone was configured from a specific catalog zone, and the zone is removed as a member from that specific catalog zone, the zone and associated state (such as zone data and DNSSEC keys) MUST be removed.
 
-[FIXME: clarify 2119 status of this and other properties and features. MUST? SHOULD? MAY? Because this one has security implications, also clearly note how ownership changes work without coo - namely, removing a member zone from catz A will cause catz B or C or D to pick up the zone on their next refresh. ]
+## Member zone name clash {#nameclash}
 
-If there is a clash between an existing member zone's name and an incoming
-member zone's name (via transfer or update), this may be an attempt to do a Change Of Ownership.
+If there is a clash between an existing zone's name (either from an existing member zone or otherwise configured zone) and an incoming
+member zone's name (via transfer or update), the new instance of the zone MUST
+be ignored and an error SHOULD be logged.
 
-A Change Of Ownership is signaled by the 'coo' property in the catalog zone currently owning the zone.
-The 'coo' property looks like "coo.<unique-N>.old.catalog PTR new.catalog".
+A clash between an existing member zone's name and an incoming member zone's name (via transfer or update), may be an attempt to migrate a zone to a different catalog.
 
-If there is no 'coo' property that resolves the clash, the zone remains owned by its current catalog and an error may be logged.
 
-If there is a 'coo' property that resolves the clash, member zone ownership is transferred to 'new.catalog'.
+## Migrating member zones between catalogs {#zonemigration}
 
-[FIXME: remove this after we agree on what IDs even mean and how zone resets work within and between catalog zones] If `<unique-N>` for the member zone is different in the new catalog zone, associated state is reset as described earlier, including existing zone data.
+If all consumers of the catalog zones involed support the `coo` property, it is RECOMMENDED to perform migration of a member zone by following the procedure described in (#cooproperty).
+Otherwise a migration of member zone from a catalog zone `$OLDCATZ` to a catalog zone `$NEWCATZ` has to be done by: first removing the member zone from `$OLDCATZ`; second adding the member zone to `$NEWCATZ` after having made sure all catalog zone consumers of `$OLDCATZ` that are also consumer of `$NEWCATZ` had the member zone removed from `$OLDCATZ`.
```

Comment:
This mandates a gap where the zone is neither in the old nor the new catalog (the member may only be created in the new catalog once removal from the old is verified). Doesn't this lead to the zone to be deprovisioned?

Reply:
Yes... to overcome you should migrate with `coo` :)